index.bs

```diff
@@ -346,10 +346,56 @@ interface mixin DatagramTransport {
 
 ## Procedures ## {#datagrams-transport-procedures}
 
-The <dfn>readDatagrams</dfn> algorithm is given a |transport| as parameter. It
-is defined by running the following steps:
-1. TODO
-1. [=ReadableStream/enqueue=] |data| in |transport|.`[[IncomingDatagrams]]`.
+To <dfn>pullDatagrams</dfn>, given a {{WebTransport}} object |transport|, run these steps:
+1. Assert: |transport|'s [=[[IncomingDatagramsPullPromise]]=] is null.
+1. Let |queue| be |transport|'s [=[[IncomingDatagramsQueue]]=].
+1. If |queue| is empty, then:
+  1. Let |promise| be a new promise.
+  1. Set |transport|'s [=[[IncomingDatagramsPullPromise]]=] to a new promise.
+  1. [=Upon fulfillment=] of |transport|'s [=[[IncomingDatagramsPullPromise]]=] with |bytes|, run
```

Comment:
Do you want to resolve this with a pair (bytes + timestamp) rather than just the bytes?

Reply:
We need the timestamp information only in _receiveDatagrams_, not in _pullDatagrams_.